.circleci/config.yml

```diff
@@ -31,3 +37,40 @@ workflows:
           filters:
             branches:
               only: master
+
+commands:
+  docker_prepare:
+    description: Downloads the buildx cli plugin and enables multiarch execution
+    parameters:
+      buildx_version:
+        type: string
+        default: "v0.4.1"
+    steps:
+      - run: apk add --no-cache curl
+      - run: mkdir -vp ~/.docker/cli-plugins/ ~/dockercache
+      - run: curl --silent -L "https://github.com/docker/buildx/releases/download/<< parameters.buildx_version >>/buildx-<< parameters.buildx_version >>.linux-amd64" > ~/.docker/cli-plugins/docker-buildx
+      - run: chmod a+x ~/.docker/cli-plugins/docker-buildx
+      - run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+      - run: docker context create old-style
+      - run: docker buildx create old-style --use
+
+  docker_build:
+    description: Builds and pushed images to dockerhub using buildx
+    parameters:
+      dockerfile:
+        type: string
+        default: docker/Dockerfile
+      push:
+        type: boolean
+        default: true
+      platforms:
+        type: string
+        default: linux/arm64/v8,linux/amd64
+      context:
+        type: string
+        default: .
+      parameters:
+        type: string
+        default: --label gitsha1=${CIRCLE_SHA1}
+    steps:
+      - run: docker buildx build -f << parameters.dockerfile >><<# parameters.push >> --push<</ parameters.push >> --platform << parameters.platforms >> << parameters.parameters >> << parameters.context >>
```

Comment:
the default output format does not work well in circleci. suggest `--progress=plain`.